### ISSUE TITLE
2D situation with missile attacks.

### ISSUE THREAD

**Issue by DimasjkTV** [suriyun-production/mmorpg-kit-docs#1443]
2D version.
The character always shoots in down direction if there are no monsters nearby, regardless of the direction.
https://youtu.be/bwuvzN-mPy8

**Comment by insthync** [comment]
Oh O_O

**Comment by RunOrDi3** [comment]
also monsters doesnt stop when hit a collider , continue walking like characters before

**Comment by insthync** [comment]
Change codes in `DamageInfoExtensions` -> `GetDamagePositionAndRotation` function to 
```
        public static void GetDamagePositionAndRotation(this IDamageInfo damageInfo, BaseCharacterEntity attacker, bool isLeftHand, AimPosition aimPosition, Vector3 stagger, out Vector3 position, out Vector3 direction, out Quaternion rotation)
        {
            if (GameInstance.Singleton.DimensionType == DimensionType.Dimension2D)
            {
                Transform damageTransform = damageInfo.GetDamageTransform(attacker, isLeftHand);
                position = damageTransform.position;
                GetDamageRotation2D(attacker.Direction2D, out rotation);
                direction = attacker.Direction2D;
                return;
            }
            if (aimPosition.type == AimPositionType.Direction)
            {
                position = aimPosition.position;
                direction = aimPosition.direction;
                rotation = Quaternion.LookRotation(direction);
            }
            else
            {
                // NOTE: Allow aim position type `None` here, may change it later
                Transform damageTransform = damageInfo.GetDamageTransform(attacker, isLeftHand);
                position = damageTransform.position;
                GetDamageRotation3D(position, aimPosition.position, stagger, out rotation);
                direction = rotation * Vector3.forward;
            }
        }
```

**Comment by insthync** [comment]
Fixed in 1.73b
